### ISSUE TITLE
@load decorator does not make new requests when params change

### ISSUE THREAD

**Issue by chuckharmston** [tonyhb/tectonic#87]
Using 2.0.2, it appears that tectonic is not recognizing that new param-affecting props in `@load`-wrapped components should be causing a new request.

I've come to that theory by adding this logging line:

```
console.log('zz', q.status, q.params);
```

After `q` is [defined here](https://github.com/tonyhb/tectonic/blob/master/packages/tectonic/src/resolver/baseResolver.js#L181), I get the following dev tools output:

<img width="247" alt="screen shot 2017-04-06 at 2 11 55 am" src="https://cloud.githubusercontent.com/assets/23885/24744219/6e3c9242-1a6e-11e7-834e-bacd72d6eb5d.png">

With this relatively simple component:

```js
@load(props => ({
  cards: Model.getList({ q: props.query })
}))
class List extends Component {
  render() {
    const { cards, status } = this.props;
    if (status.cards.isPending()) {
      return <p>Loading...</p>;
    } else if (status.cards.isSuccess()) {
      return (
        <ol>
            {cards.map(card => <p key={card.id}>{card.id}</p>)}
        </ol>
      );
    }
    console.log(status);
    return <p>Error</p>;
  }
}

```

The first render works perfectly. When the props change, a new request is not made, but the wrapped component's `render` method is called. In that case, the `status` prop looks like:

```js
{
  cards: {
    code: undefined,
    error: undefined
  }
}
```

**Comment by tonyhb** [comment]
Thanks for the report!

A little background on how this is probably being lost:

React re-evaluates a component many times.  Unfortunately for us, we decided to build caching and manage stale data as a first class activity.  This means that if your API isn't allowing us to cache our data we need to be able to _not_ re-query for data every time React re-evaluates a `@load` component.

So, we store an internal status on the query which, when set, skips resolution.

This is probably stale with new props.

I'll take a look and add some test cases for this and hopefully come up with a fix soon!

**Comment by tonyhb** [comment]
Fixed in #103
